src/shared_realm.cpp

```diff
@@ -666,16 +717,19 @@ uint64_t Realm::get_schema_version(const Realm::Config &config)
 
 void Realm::close()
 {
+    // Order matters here: if we're the last Realm instance for this path, we
+    // need to tear down the coordinator before destroying our shared group
+    // to maintain the coordinator's invariant that there be a continuous session
```

Comment:
With out-of-file transaction logs the transaction version would reset if the sharedgroup is closed entirely and reopened, which isn't handled by this code. With in-realm transaction logs the version is persisted, so I guess this no longer applies.